src/linux/Packaging.Linux/build.sh

```diff
@@ -135,61 +151,74 @@ mv "$PAYLOAD"/*.pdb "$SYMBOLOUT" || exit 1
 echo "Setting file permissions..."
 /bin/chmod -R 755 "$PAYLOAD" || exit 1
 
-# Build binaries tarball
-echo "Building binaries tarball..."
-pushd "$PAYLOAD"
-tar -czvf "$TARBALL" * || exit 1
-popd
-
-# Build symbols tarball
-echo "Building symbols tarball..."
-pushd "$SYMBOLOUT"
-tar -czvf "$SYMTARBALL" * || exit 1
-popd
-
-# Build .deb
-INSTALL_TO="$DEBROOT/usr/local/share/gcm-core/"
-LINK_TO="$DEBROOT/usr/local/bin/"
-mkdir -p "$DEBROOT/DEBIAN" "$INSTALL_TO" "$LINK_TO" || exit 1
-
-# make the debian control file
-cat >"$DEBROOT/DEBIAN/control" <<EOF
-Package: gcmcore
-Version: $VERSION
-Section: vcs
-Priority: optional
-Architecture: $ARCH
-Depends:
-Maintainer: GCM <gitfundamentals@github.com>
-Description: Cross Platform Git Credential Manager command line utility.
- GCM supports authentication with a number of Git hosting providers
- including GitHub, BitBucket, and Azure DevOps.
- For more information see https://aka.ms/gcm
+if [ "$INSTALL_FROM_SOURCE" = true ]; then
+    echo "Installing..."
+
+    # Install directories
+    INSTALL_TO="$INSTALL_LOCATION/share/gcm-core/"
+    LINK_TO="$INSTALL_LOCATION/bin/"
+else
+    echo "Packing Packaging.Linux..."
+    # Cleanup any old archive files
+    if [ -e "$TAROUT" ]; then
+        echo "Deleteing old archive '$TAROUT'..."
+        rm "$TAROUT"
+    fi
+
+    # Ensure the parent directory for the archive exists
+    mkdir -p "$TAROUT" || exit 1
+
+    # Build binaries tarball
+    echo "Building binaries tarball..."
+    pushd "$PAYLOAD"
+    tar -czvf "$TARBALL" * || exit 1
+    popd
+
+    # Build symbols tarball
+    echo "Building symbols tarball..."
+    pushd "$SYMBOLOUT"
+    tar -czvf "$SYMTARBALL" * || exit 1
+    popd
+
+    # Build .deb
+    INSTALL_TO="$DEBROOT/usr/local/share/gcm-core/"
+    LINK_TO="$DEBROOT/usr/local/bin/"
+
+    # make the debian control file
+    cat >"$DEBROOT/DEBIAN/control" <<EOF
+    Package: gcmcore
+    Version: $VERSION
+    Section: vcs
+    Priority: optional
+    Architecture: $ARCH
+    Depends:
+    Maintainer: GCM <gitfundamentals@github.com>
+    Description: Cross Platform Git Credential Manager command line utility.
+    GCM supports authentication with a number of Git hosting providers
+    including GitHub, BitBucket, and Azure DevOps.
+    For more information see https://aka.ms/gcm
 EOF
 
+    dpkg-deb --build "$DEBROOT" "$DEBPKG" || exit 1
+
+    echo "Pack complete."
+fi
+
+mkdir -p "$INSTALL_TO" "$LINK_TO" || exit 1
+
 # Copy all binaries and shared libraries to target installation location
 cp -R "$PAYLOAD"/* "$INSTALL_TO" || exit 1
 
 # Create symlink
-ln -s -r "$INSTALL_TO/git-credential-manager-core" \
-    "$LINK_TO/git-credential-manager-core" || exit 1
-
-dpkg-deb --build "$DEBROOT" "$DEBPKG" || exit 1
+if [ ! -f "$LINK_TO/git-credential-manager-core" ]; then
+    ln -s -r "$INSTALL_TO/git-credential-manager-core" \
+        "$LINK_TO/git-credential-manager-core" || exit 1
+fi
 
-echo "Pack complete."
+echo "Install complete."
```

Comment:
I guess this should still say `Pack complete` if `"$INSTALL_FROM_SOURCE" != true`? The way I would do it: set a variable `message` in the `if`/`else` above, to `"Install complete."` and `"Pack complete."`, respectively.